runtime/tests/stake.rs

```diff
@@ -32,45 +32,36 @@ use {
 };
 
 /// get bank at next epoch + `n` slots
-fn next_epoch_and_n_slots(bank: &Arc<Bank>, mut n: usize) -> Arc<Bank> {
+fn next_epoch_and_n_slots(bank: Arc<Bank>, mut n: usize) -> Arc<Bank> {
     bank.squash();
-    let mut bank = Arc::new(Bank::new_from_parent(
-        bank,
-        &Pubkey::default(),
-        bank.get_slots_in_epoch(bank.epoch()) + bank.slot(),
-    ));
+    let slot = bank.get_slots_in_epoch(bank.epoch()) + bank.slot();
+    let mut bank = Arc::new(Bank::new_from_parent(bank, &Pubkey::default(), slot));
 
     while n > 0 {
         bank.squash();
-        bank = Arc::new(Bank::new_from_parent(
-            &bank,
-            &Pubkey::default(),
-            1 + bank.slot(),
-        ));
+        let slot = bank.slot() + 1;
+        bank = Arc::new(Bank::new_from_parent(bank, &Pubkey::default(), slot));
         n -= 1;
     }
 
     bank
 }
 
 fn fill_epoch_with_votes(
-    bank: &Arc<Bank>,
+    bank: Arc<Bank>,
```

Comment:
just make this `mut` instead of creating a `mut` shadow bank?

Reply:
https://github.com/solana-labs/solana/pull/32802/commits/a0abd278a923261c06d43c31624df93d78b5baa9#diff-f381ab88d121890895f1c4e1a5a77157fcb0d11742509645d87d50f9944987d5R51